src/rgw/rgw_rados.cc

```diff
@@ -1492,7 +1492,7 @@ int RGWZoneParams::fix_pool_names()
   }
 
   domain_root = fix_zone_pool_name(pool_names, name, ".rgw.data.root", domain_root.name);
-  metadata_heap = fix_zone_pool_name(pool_names, name, ".rgw.meta", metadata_heap.name);
+  metadata_heap = "";
```

Comment:
in testing, i see that this does work to turn off the metadata_heap by default. however, it also prevents us from using `zone set --infile zone.json` to turn it back on - we decode the new value for `metadata_heap`, then call `fix_pool_names()` and clear it again. i was able to get the `zone set` to work with the following change:
```diff
-  metadata_heap = fix_zone_pool_name(pool_names, name, ".rgw.meta", metadata_heap.name);
+  if (!metadata_heap.name.empty()) { // optional
+    metadata_heap = fix_zone_pool_name(pool_names, name, ".rgw.meta", metadata_heap.name);
+  }
```